proof/refine/RISCV64/Ipc_R.thy

```diff
@@ -2176,13 +2176,66 @@ lemma refillSufficient_sp:
    \<lbrace>\<lambda>rv s. P s \<and> (\<exists>sc. scs_of' s scp = Some sc \<and> rv = (refillSufficient usage (refillHd sc)))\<rbrace>"
   by wpsimp
 
+lemma cap_in_tcbFaultHandlerSlot:
+  "\<lbrakk>ko_at' (tcb :: tcb) t s; valid_tcbs' s\<rbrakk>
+   \<Longrightarrow> \<exists>cap. cte_wp_at' (\<lambda>c. cteCap c = cap) (t + 2 ^ cte_level_bits * tcbFaultHandlerSlot) s
+             \<and> valid_cap' cap s
+             \<and> (\<exists>n\<in>dom tcb_cte_cases. cte_wp_at' (\<lambda>cte. cteCap cte = cap) (t + n) s)"
```

Comment:
I have the feeling that `t + 2 ^ cte_level_bits * tcbFaultHandlerSlot` has a name somewhere, or at least it should. It may be written in a slightly different form (e.g. `1 << _`), or in the worst case even expanded. Don't quite remember where, though. I think @Xaphiosis encountered these recently in arch-splitting Refine.

Reply:
In `tcb_cte_cases_simps`, we have something like this on master (as part of the new arch-split and `tcb_cte_cases_neqs` infrastructure):
`tcb_cte_cases (3 << cteSizeBits) = Some (tcbCaller, tcbCaller_update)`
so I expect that the compressed version for this case might be:
`t + (tcbFaultHandlerSlot << cteSizeBits)`? in fact, something like this shows up below